### PR TITLE
Stop executing testcases when proto limit exceeded

### PR DESCRIPTION
**Pull request by IvanBM18**
Bug:b/473624472

Enforces protobuf message size limits (PROTOBUF_MSG_LIMIT) during blackbox testcase execution in fuzz_task.py.

## Context
In the previous PR in the stack, we changed so that in utask_main we now save the fuzzer logs in a protobuf message, which is consumed  at `postprocess`.

**Previously:**
For each test case in BlackBox fuzzing, we immediately uploaded its fuzz & output logs  into GCS. This proved basically no limitations.

**Now:**
For each test case in BlackBox fuzzing, we save its fuzz & output logs into a protobuf.

The protobuf has a hard limit of 2GB(see[ the docs](https://protobuf.dev/programming-guides/proto-limits/)).

## Overview
To prevent a possible edge case in which logs + output from a big fuzzing session surpasses the protobuf size limit, we now limit the test cases executed in the session once they trespass the protobuf limit size.

## Changes
* **Early Loop Termination:** Checks if `fuzz_task_output` size exceeds 1.8Gb after each batch of worker threads and terminates the execution loop early in case it does.
* **Output Limit Guard in Queue Processing:** Prevents further `fuzzer_run_output `entries from being appended once the 1.8Gb is exceeded, while still preserving all discovered crashes.
* **Clean Unexecuted Testcase Handling:** Deletes unexecuted testcase files from disk and slices returned testcase paths and metadata to ensure metrics (testcases_executed) report accurately.
* **Unit Tests:** Adds unit tests for a bunch of edge cases(theres actually more test cases than actual prod code).

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/testcase_manager.py [hunk 1]
```diff
@@ -586,7 +586,7 @@ def _do_run_testcase_and_return_result_in_queue(
 
     # Save raw, unsymbolized execution output for deferred postprocessing.
     crash_result_full = CrashResult(return_code, crash_time, output)
-    unsymbolized_output = crash_result_full.get_stacktrace(symbolized=False)
+    unsymbolized_output = crash_result_full.get_stacktrace()
```
Comment (IvanBM18) [inline src/clusterfuzz/_internal/bot/testcase_manager.py:594]: I added the `symbolized=False` for some tests on other PR, but i forgot to turn it back to the way it was: `crash_result_full.get_stacktrace()`
So i'm just adding it back here.

File: src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py
Comment (dylanjew) [inline src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py:1857]: here, this won't be accurate. Although, i think it wasn't always accurate to begin with if there was an error at some point too.

File: src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py [hunk 2]
```diff
@@ -1928,8 +1973,13 @@ def do_blackbox_fuzzing(self, fuzzer, fuzzer_directory, job_type):
       crashes = [
           Crash.from_testcase_manager_crash(crash) for crash in crashes if crash
       ]
-    return (generate_result.fuzzer_metadata, testcase_file_paths,
-            testcases_metadata, crashes)
+    executed_testcase_file_paths, executed_testcases_metadata = (
```
Comment (dylanjew) [inline src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py]: it might be clearer to have this only clean up unexecuted testcases if `test_number < testcases_generated`

File: src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/fuzz_task_test.py [hunk 3]
```diff
@@ -1552,6 +1590,134 @@ def test_add_crashes_from_temp_queue(self):
...
+        fuzz_task_input=uworker_msg_pb2.FuzzTaskInput())
+    return fuzz_task.FuzzingSession(uworker_input, 10)
+
+  def test_is_fuzz_task_output_limit_exceeded_below_limit(self):
+    """Verify _is_fuzz_task_output_limit_exceeded returns False below PROTOBUF_MSG_LIMIT."""
+    session = self._create_session()
+    self.assertFalse(session._is_fuzz_task_output_limit_exceeded())
+
+  @mock.patch(
+      'clusterfuzz._internal.bot.tasks.utasks.fuzz_task.PROTOBUF_MSG_LIMIT', 0)
```
Comment (dylanjew) [inline src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/fuzz_task_test.py]: nit: it's probably more realistic to set or mock `self.fuzz_task_output.ByteSize()` rather than just setting the `PROTOBUF_MSG_LIMIT` to 0

File: src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py (hunk 2 above)
Comment (IvanBM18) [reply src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py]: Good point, thanks!

File: src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/fuzz_task_test.py (hunk 3 above)
Comment (IvanBM18) [reply src/clusterfuzz/_internal/tests/core/bot/tasks/utasks/fuzz_task_test.py]: Done!

File: src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py
Comment (IvanBM18) [reply src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py:1857]: You mean that we should update `self.fuzz_task_output` at the end of the fuzzing session so that it has the total of executed test cases instead?
Or WDYM by it being innacuarate?

Comment (dylanjew) [reply src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py:1857]: Oh, sorry I was confused. I thought this was the number of testcases executed that we were reporting, and that it would be inaccurate because we are now stopping early.

We do that correctly here: https://github.com/google/clusterfuzz/blob/b26420660bf6da662227a5df1c30a4fd06044c12/src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py#L2061

Comment (dylanjew) [inline src/clusterfuzz/_internal/bot/tasks/utasks/fuzz_task.py:1834]: this was what I got confused with. It's a known bug that we don't actually emit the correct testcase count when `generate_blackbox_testcases` doesn't actually generate the max amount of testcases for whatever reason.

**Review by dylanjew — APPROVED** [review]

**Review by javanlacerda — APPROVED** [review]
